driver/razeraccessory_driver.c

```diff
@@ -695,7 +743,13 @@ static ssize_t razer_attr_write_matrix_effect_wave(struct device *dev, struct de
         // Direction values are flipped compared to other devices
         direction ^= ((1<<0) | (1<<1));
         request = razer_chroma_extended_matrix_effect_wave(VARSTORE, ZERO_LED, direction);
-        request.transaction_id.id = 0x1F;
+
+        if(device->usb_dev->descriptor.idProduct == USB_DEVICE_ID_RAZER_MOUSE_DOCK_PRO) {
+            request.transaction_id.id = 0xFF;
+        } else {
+            request.transaction_id.id = 0x1F;
+        }
```

Comment:
I'm not happy about this extra if, but how the code is currently structured, I guess it doesn't become much better than that...